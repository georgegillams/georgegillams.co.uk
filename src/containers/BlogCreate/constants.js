export const KEY = 'blog-list';
export const CREATE_BLOG = `app/${KEY}/CREATE_BLOG`;
