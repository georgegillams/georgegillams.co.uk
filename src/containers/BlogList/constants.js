export const KEY = 'blog-list';
export const LOAD_BLOGS = `app/${KEY}/LOAD_BLOGS`;
export const DELETE_BLOG = `app/${KEY}/DELETE_BLOG`;
