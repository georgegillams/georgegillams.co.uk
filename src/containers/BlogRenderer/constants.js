export const KEY = 'blog-renderer';
export const LOAD_BLOG = `app/${KEY}/LOAD_BLOG`;
export const UPDATE_BLOG = `app/${KEY}/UPDATE_BLOG`;
