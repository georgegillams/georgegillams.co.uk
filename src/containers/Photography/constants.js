export const KEY = 'photography';
export const LOAD_PHOTOS = `app/${KEY}/LOAD_PHOTOS`;
