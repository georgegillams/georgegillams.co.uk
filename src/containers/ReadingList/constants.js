export const KEY = 'reading-list';
export const LOAD_BOOKS = `app/${KEY}/LOAD_BOOKS`;
export const UPDATE_BOOK = `app/${KEY}/UPDATE_BOOK`;
export const DELETE_BOOK = `app/${KEY}/DELETE_BOOK`;
