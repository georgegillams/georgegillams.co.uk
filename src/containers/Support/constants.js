export const KEY = 'support';
export const LOAD_LINKS = `app/${KEY}/LOAD_LINKS`;
export const CREATE_LINK = `app/${KEY}/CREATE_LINK`;
export const DELETE_LINK = `app/${KEY}/DELETE_LINK`;
