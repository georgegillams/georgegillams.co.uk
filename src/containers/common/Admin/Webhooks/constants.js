export const KEY = 'webhooks';

export const LOAD_ENDPOINTS = `app/${KEY}/LOAD_ENDPOINTS`;
export const CREATE_ENDPOINT = `app/${KEY}/CREATE_ENDPOINT`;
export const UPDATE_ENDPOINT = `app/${KEY}/UPDATE_ENDPOINT`;
export const REMOVE_ENDPOINT = `app/${KEY}/REMOVE_ENDPOINT `;

export const LOAD_NOTIFICATIONS = `app/${KEY}/LOAD_NOTIFICATIONS`;
export const REMOVE_NOTIFICATION = `app/${KEY}/REMOVE_NOTIFICATION`;
