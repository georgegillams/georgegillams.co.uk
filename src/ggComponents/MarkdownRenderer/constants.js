const DEFAULT_SUPPORTED_FEATURES = [
  'code',
  'blockCode',
  'bold',
  'italic',
  'strikethrough',
  'quotation',
  'bigLink',
  'link',
  'footnote',
  'image',
  'video',
  'citation',
  'sectioning',
  'linebreak',
  'bulletList',
  'numberedList',
];

export default DEFAULT_SUPPORTED_FEATURES;
export { DEFAULT_SUPPORTED_FEATURES };
