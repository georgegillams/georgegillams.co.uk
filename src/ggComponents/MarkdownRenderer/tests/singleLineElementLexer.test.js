/* eslint-disable max-len */

import {
  parseSingleLineElements,
  parseStringForBoldness,
  parseStringForCitations,
  parseStringForCode,
  parseStringForFootnote,
  parseStringForFootnoteReference,
  parseStringForImage,
  parseStringForItalic,
  parseStringForLink,
  parseStringForBigLink,
  parseStringForReferences,
  parseStringForYouTube,
  parseStringForSmartImage,
  parseStringForStrikethrough,
  parseStringForTripleStarBoldItalic,
} from '../singleLineElementLexer.js';
import { DEFAULT_SUPPORTED_FEATURES } from '../constants.js';

const simpleFurtherProcess = x => `forFurtherProcessing: "${x}"`;

// #region individual items
test('parses text containing image', () => {
  const result = parseStringForImage('Line containing ![imageCaption](imageSrc) text', simpleFurtherProcess);
  expect(result.error).toBe(undefined);
  expect(result).toMatchSnapshot();
});
test('parses text containing big link', () => {
  const result = parseStringForBigLink(`*[Big link to DDG](https://duckduckgo.com/)`, simpleFurtherProcess);
  expect(result.error).toBe(undefined);
  expect(result).toMatchSnapshot();
});
test('parses text containing link', () => {
  const result = parseStringForLink(
    `This contains a [link to DDG](https://duckduckgo.com/) text.`,
    simpleFurtherProcess
  );
  expect(result.error).toBe(undefined);
  expect(result).toMatchSnapshot();
});
test('parses text containing strikethrough', () => {
  const result = parseStringForStrikethrough(`This contains some ~stricken~ text.`, simpleFurtherProcess);
  expect(result.error).toBe(undefined);
  expect(result).toMatchSnapshot();
});
test('parses text containing footnote', () => {
  const result = parseStringForFootnote(
    `This footnote[^1] points to something under the paragraph.\nThis is the end of the paragraph\n[^1]: This is **the** footnote`,
    simpleFurtherProcess
  );
  expect(result.error).toBe(undefined);
  expect(result).toMatchSnapshot();
});
test('parses text containing footnote reference', () => {
  const result = parseStringForFootnoteReference(
    `This footnote[^1] points to something under the paragraph.\nThis is the end of the paragraph\n[^1]: This is **the** footnote`,
    simpleFurtherProcess
  );
  expect(result.error).toBe(undefined);
  expect(result).toMatchSnapshot();
});
test('parses text containing citation', () => {
  const result = parseStringForCitations(
    `This points to some citation!cite(7) that will appear further down!cite(10) the page.`,
    simpleFurtherProcess
  );
  expect(result.error).toBe(undefined);
  expect(result).toMatchSnapshot();
});
test('parses text containing reference', () => {
  const result = parseStringForReferences(
    `!reference(4): This is the forth reference\n!reference(5): This is the fifth reference`,
    simpleFurtherProcess
  );
  expect(result.error).toBe(undefined);
  expect(result).toMatchSnapshot();
});
test('parses text containing smart image', () => {
  const result = parseStringForSmartImage(
    'Line containing ![1x2][imageCaption](lightImageSrc)(darkImageSrc) followed by some **emphasized** text',
    simpleFurtherProcess
  );
  expect(result.error).toBe(undefined);
  expect(result).toMatchSnapshot();
});
test('parses text containing video', () => {
  const result = parseStringForYouTube(
    'Line containing !yt[true](jofNR_WkoCE) followed by some **emphasized** text',
    simpleFurtherProcess
  );
  expect(result.error).toBe(undefined);
  expect(result).toMatchSnapshot();
});
test('parses text containing italic', () => {
  const result = parseStringForItalic(`Line containing _wonky_ text\nAnd this is *some more*.`, simpleFurtherProcess);
  expect(result.error).toBe(undefined);
  expect(result).toMatchSnapshot();
});
test('parses text containing code', () => {
  const result = parseStringForCode('Line containing `some inline code` just like that', simpleFurtherProcess);
  expect(result.error).toBe(undefined);
  expect(result).toMatchSnapshot();
});
test('parses text containing bold', () => {
  const result = parseStringForBoldness('Line containing **emphasized** text', simpleFurtherProcess);
  expect(result.error).toBe(undefined);
  expect(result).toMatchSnapshot();
});
test('parses text containing 3-star bold italic', () => {
  const result = parseStringForTripleStarBoldItalic('Line containing ***emphasized*** text', simpleFurtherProcess);
  expect(result.error).toBe(undefined);
  expect(result).toMatchSnapshot();
});
test('parses text containing bold italic', () => {
  const result = parseStringForBoldness('Line containing **_emphasized_** text', simpleFurtherProcess);
  expect(result.error).toBe(undefined);
  expect(result).toMatchSnapshot();
});
// #endregion

// #region structure
test('parses paragraphs containing styled text', () => {
  const result = parseSingleLineElements(
    [
      { type: 'paragraph', children: ['Line containing **emphasized** text'] },
      { type: 'paragraph', children: ['Line containing _wonky_ text'] },
      {
        type: 'paragraph',
        children: ['Line containing [link to DDG](https://duckduckgo.com/)'],
      },
    ],
    DEFAULT_SUPPORTED_FEATURES
  );
  expect(result.error).toBe(undefined);
  expect(result).toMatchSnapshot();
});
// #endregion
