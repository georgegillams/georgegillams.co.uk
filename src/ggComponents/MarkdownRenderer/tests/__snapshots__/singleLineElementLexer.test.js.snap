// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parses paragraphs containing styled text 1`] = `
Array [
  Object {
    "children": Array [
      "Line containing ",
      Object {
        "children": Array [
          "emphasized",
        ],
        "type": "bold",
      },
      " text",
    ],
    "type": "paragraph",
  },
  Object {
    "children": Array [
      "Line containing ",
      Object {
        "children": Array [
          "wonky",
        ],
        "type": "italic",
      },
      " text",
    ],
    "type": "paragraph",
  },
  Object {
    "children": Array [
      "Line containing ",
      Object {
        "children": Array [
          "link to DDG",
        ],
        "href": "https://duckduckgo.com/",
        "hrefExternal": true,
        "type": "link",
      },
      "",
    ],
    "type": "paragraph",
  },
]
`;

exports[`parses text containing 3-star bold italic 1`] = `
Array [
  "forFurtherProcessing: \\"Line containing \\"",
  Object {
    "children": Array [
      Object {
        "children": Array [
          "emphasized",
        ],
        "type": "italic",
      },
    ],
    "type": "bold",
  },
  "forFurtherProcessing: \\" text\\"",
]
`;

exports[`parses text containing big link 1`] = `
Array [
  Object {
    "href": "https://duckduckgo.com/",
    "hrefExternal": true,
    "text": "Big link to DDG",
    "type": "bigLink",
  },
]
`;

exports[`parses text containing bold 1`] = `
Array [
  "forFurtherProcessing: \\"Line containing \\"",
  Object {
    "children": Array [
      "emphasized",
    ],
    "type": "bold",
  },
  "forFurtherProcessing: \\" text\\"",
]
`;

exports[`parses text containing bold italic 1`] = `
Array [
  "forFurtherProcessing: \\"Line containing \\"",
  Object {
    "children": Array [
      "_emphasized_",
    ],
    "type": "bold",
  },
  "forFurtherProcessing: \\" text\\"",
]
`;

exports[`parses text containing citation 1`] = `
Array [
  "forFurtherProcessing: \\"This points to some citation\\"",
  Object {
    "identifier": "7",
    "type": "citation",
  },
  "forFurtherProcessing: \\" that will appear further down!cite(10) the page.\\"",
]
`;

exports[`parses text containing code 1`] = `
Array [
  "forFurtherProcessing: \\"Line containing \\"",
  Object {
    "text": "some inline code",
    "type": "code",
  },
  "forFurtherProcessing: \\" just like that\\"",
]
`;

exports[`parses text containing footnote 1`] = `
Array [
  Object {
    "children": Array [
      "This is **the** footnote",
    ],
    "number": "1",
    "type": "footnote",
  },
]
`;

exports[`parses text containing footnote reference 1`] = `
Array [
  "forFurtherProcessing: \\"This footnote\\"",
  Object {
    "number": "1",
    "type": "footnoteReference",
  },
  "forFurtherProcessing: \\" points to something under the paragraph.
This is the end of the paragraph
[^1]: This is **the** footnote\\"",
]
`;

exports[`parses text containing image 1`] = `
Array [
  "forFurtherProcessing: \\"Line containing \\"",
  Object {
    "caption": "imageCaption",
    "src": "imageSrc",
    "type": "image",
  },
  "forFurtherProcessing: \\" text\\"",
]
`;

exports[`parses text containing italic 1`] = `
Array [
  "forFurtherProcessing: \\"Line containing \\"",
  Object {
    "children": Array [
      "wonky",
    ],
    "type": "italic",
  },
  "forFurtherProcessing: \\" text
And this is *some more*.\\"",
]
`;

exports[`parses text containing link 1`] = `
Array [
  "forFurtherProcessing: \\"This contains a \\"",
  Object {
    "children": Array [
      "link to DDG",
    ],
    "href": "https://duckduckgo.com/",
    "hrefExternal": true,
    "type": "link",
  },
  "forFurtherProcessing: \\" text.\\"",
]
`;

exports[`parses text containing reference 1`] = `
Array [
  Object {
    "identifier": "4",
    "reference": "This is the forth reference
!reference(5): This is the fifth reference",
    "type": "reference",
  },
]
`;

exports[`parses text containing smart image 1`] = `
Array [
  "forFurtherProcessing: \\"Line containing \\"",
  Object {
    "aspectX": "1",
    "aspectY": "2",
    "caption": "imageCaption",
    "darkSrc": "darkImageSrc",
    "lightSrc": "lightImageSrc",
    "type": "smartImage",
  },
  "forFurtherProcessing: \\" followed by some **emphasized** text\\"",
]
`;

exports[`parses text containing strikethrough 1`] = `
Array [
  "forFurtherProcessing: \\"This contains some \\"",
  Object {
    "children": Array [
      "stricken",
    ],
    "type": "strikethrough",
  },
  "forFurtherProcessing: \\" text.\\"",
]
`;

exports[`parses text containing video 1`] = `
Array [
  "forFurtherProcessing: \\"Line containing \\"",
  Object {
    "showSuggestions": true,
    "type": "youtube",
    "videoID": "jofNR_WkoCE",
  },
  "forFurtherProcessing: \\" followed by some **emphasized** text\\"",
]
`;
